python/tskit/genotypes.py

```diff
@@ -215,13 +225,55 @@ def copy(self) -> Variant:
         """
         Create a copy of this Variant. Note that calling :meth:`decode` on the
         copy will fail as it does not take a copy of the internal tree.
+
         :return: The copy of this Variant.
         """
         variant_copy = Variant.__new__(Variant)
         variant_copy.tree_sequence = self.tree_sequence
         variant_copy._ll_variant = self._ll_variant.restricted_copy()
         return variant_copy
 
+    def counts(self) -> dict[str, int]:
+        """
+        Return a count of the number of samples possessing each allele. Missing
+        data is represented by an allelic state of ``None``.
+
+        :return: A dictionary mapping each allelic state to the number of samples
+            with that allele.
+        """
+        counts = {a: 0 for a in self.alleles}
+        for g, count in zip(*np.unique(self.genotypes, return_counts=True)):
+            counts[self.alleles[g]] = count
+        return counts
+
+    def frequencies(self, remove_missing=None) -> dict[str, float]:
+        """
+        Return the frequency of each allele at this site (i.e. the number of samples
+        with each allele, divided by the total number of samples).
+
+        :param bool remove_missing: If True, only samples with non-missing data will
+            be counted in the total number of samples used to calculate the frequency,
+            and no information on the frequency of missing data is returned. Otherwise
+            (default), samples with missing data are included when calculating
+            frequencies.
+        :return: A dictionary mapping allelic states to the frequency of each allele
+            among the samples
+        """
+        if remove_missing is None:
+            remove_missing = False
+        total = self.tree_sequence.num_samples
```

Comment:
Pretty sure this should be `len(self.samples)`?

Reply:
Hmm, that's an interesting question. If you call `ts.variants(samples=[0,1])`, do you expect the allele frequencies to be calculated from the entire tree sequence, or just among the samples you have specified. I'm not sure?

Reply:
No, I'm being stupid here. The counts are only among the specified samples, so you are right @benjeffery

Reply:
If I've specified `samples` to `Variant` then I would expect that all its operations are relative to that set of samples. Instead of `report_local_frequencies` how about  `denominator` which defaults to `len(self.samples)`?